build.gradle

```diff
@@ -81,19 +87,92 @@ subprojects {
 def moduleVersion = project.version.replace("-SNAPSHOT", "")
 
 release {
+    // Disable check snapshots temporarily
     failOnPublishNeeded = false
-    failOnSnapshotDependencies = true
-
-    buildTasks = ['build']
     versionPropertyFile = 'gradle.properties'
-    tagTemplate = 'v$version'
-
+    tagTemplate = 'v${version}'
     git {
+        // To release from any branch
         requireBranch = "release-${moduleVersion}"
         pushToRemote = 'origin'
     }
 }
 
-task build {
-    dependsOn('persist-cli:build')
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+task codeCoverageReport(type: JacocoReport) {
+    dependsOn = subprojects.test
+
+    executionData fileTree(project.rootDir.absolutePath).include("**/build/coverage-reports/*.exec")
+
+    subprojects.each {
+        sourceSets it.sourceSets.main
+    }
+    reports {
+        xml.enabled(true)
+        html.enabled(true)
+        csv.enabled(true)
+        xml.destination(new File("${buildDir}/reports/jacoco/report.xml"))
+        html.destination(new File("${buildDir}/reports/jacoco/report.html"))
+        csv.destination(new File("${buildDir}/reports/jacoco/report.csv"))
+    }
+
+    onlyIf = {
+        true
+    }
+}
+
+def artifactCacheParent = file("$project.projectDir/build/cache_parent/")
+def artifactLibParent = file("$project.projectDir/build/lib_parent/")
+def targetPersistCliJar = file("$project.rootDir/persist-cli/build/libs/persist-cli-${project.version}.jar")
+
+jar {
+    manifest {
+        attributes('Implementation-Title': project.name,
+                'Implementation-Version': project.version)
+    }
+}
+
+task directoryBuild {
+    inputs.dir file(project.projectDir)
+    doLast {
+        copy {
+            from targetPersistCliJar
+            into file("$artifactLibParent/libs")
+        }
+    }
+
+    outputs.dir artifactCacheParent
+    outputs.dir artifactLibParent
+}
+
+task createArtifactZip(type: Zip) {
+    destinationDirectory = file("${buildDir}/distributions")
+    from directoryBuild
 }
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            artifact source: createArtifactZip, extension: 'zip'
+        }
+    }
+
+    repositories {
+        maven {
+            name = "GitHubPackages"
+            url = uri("https://maven.pkg.github.com/ballerina-platform/persist-tools")
+            credentials {
+                username = System.getenv("packageUser")
```

Comment:
Change this to publish User and publish PAT. Else anyone with access can accidently publish